### ISSUE TITLE
Miprov2 doesn't work with TypedPredictors

### ISSUE THREAD

**Issue by eelang-sc** [stanfordnlp/dspy#1295]
https://github.com/stanfordnlp/dspy/blob/d8b8909773fc31e72cec093db2f26109590e524e/dspy/propose/grounded_proposer.py#L198

As the TODO line suggests it doesn't work with typed predictors.

a quick fix for me was to change
```
pattern = r"^(.*dspy\.(ChainOfThought|Predict).*)$"  # TODO: make it so that this extends out to any dspy Module
```
to
```
pattern = r"^(.*dspy\.(ChainOfThought|Predict|TypedPredictor).*)$"  # TODO: make it so that this extends out to any dspy Module
```

**Comment by okhat** [comment]
good to get this on the radar of @XenonMolecule

**Comment by haznai** [comment]
Proposed a trivial fix for this suggested change in #1453 


Thanks for your time and effort, people :)

**Comment by XenonMolecule** [comment]
Thanks @haznai this looks great!  I think this should work for ReAct and ProgramOfThought too
